frame/staking-rewards/src/lib.rs

```diff
@@ -599,8 +626,118 @@ pub mod pallet {
 					.try_insert(*asset_id, inflation)
 					.map_err(|_| Error::<T>::ReductionConfigProblem)?;
 			}
+
 			Ok((rewards_btree_map, reductions))
 		}
+
+		pub(crate) fn reward_acumulation_hook_reward_update_calculation(
+			pool_id: T::RewardPoolId,
+			reward: Reward<T::AssetId, T::Balance>,
+			now: u64,
+		) -> Reward<T::AssetId, T::Balance> {
+			match now.safe_sub(&reward.last_updated_timestamp) {
+				Ok(elapsed_time) => {
+					let reward_rate_period_seconds = match reward.reward_rate.period {
+						RewardRatePeriod::PerSecond => 1,
+					};
+
+					// SAFETY(benluelo): Usage of Div::div:
+					//
+					// Integer division can only fail if rhs == 0, and
+					// reward_rate_period_seconds is non-zero here as defined above.
+					let periods_surpassed = elapsed_time.div(reward_rate_period_seconds);
+
+					if periods_surpassed.is_zero() {
+						reward
+					} else {
+						let new_total_rewards = u128::from(periods_surpassed)
+							.saturating_mul(reward.reward_rate.amount.into())
+							.saturating_add(reward.total_rewards.into());
+
+						let new_total_rewards = if new_total_rewards <= reward.max_rewards.into() {
+							new_total_rewards.into()
+						} else {
+							// saturate at max_rewards, but emit an error first
+							// only emit if the previous total_rewards wasn't at the max_rewards
+							// (i.e. the event would have been emitted already)
+							if reward.total_rewards < reward.max_rewards {
+								Self::deposit_event(Event::<T>::MaxRewardsAccumulated {
+									pool_id,
+									asset_id: reward.asset_id,
+								});
+							}
+							reward.max_rewards
+						};
+
+						Reward {
+							total_rewards: new_total_rewards,
+							last_updated_timestamp: reward
+								.last_updated_timestamp
```

Comment:
Because there could be a situation where the reward period doesn't line up with the block time (think a period of 5 seconds, and a block time of 6 seconds); after 5 blocks (30 secs) you would expect the rewards to have accumulated 6 times, but it would only be 5 since 1 second would be "lost" each block

Reply:
This was more of an issue when working with arbitrary reward periods, but I think it's good to keep in for if/when we add more reward periods

Reply:
(basically the issue is `let periods_surpassed = elapsed_time.div(reward_rate_period_seconds);` is floor division)